Core.Collectors/Collector/CachingCollector.cs

```diff
@@ -36,7 +42,7 @@ public async Task<bool> ProcessAndCacheAsync(TCollectionNode collectionNode, TEn
                 if (cachedProgressRecord != null && cachedProgressRecord.Succeeded)
                 {
                     // If the cache includes the progress record and record.Succeeded is true, skip re-collection since this endpoint was previously collected successfully.
-
+                    TCollectionNode collectionNode = collectionNodes.First();
```

Comment:
There is no check that the collection nodes contain any elements, is there? Thus, calling First() might cause issues.